Packages/com.chisel.graph/Chisel/ChiselGraphInstance.cs

```diff
@@ -0,0 +1,131 @@
+using Chisel.Core;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Unity.Collections;
+using Unity.Jobs;
+using UnityEngine;
+
+namespace Chisel.Nodes
+{
+    [ExecuteInEditMode]
+    public class ChiselGraphInstance : MonoBehaviour
+    {
+        public ChiselGraph graph;
+        public MeshFilter meshFilter;
+        public MeshRenderer meshRenderer;
+        public bool IsDirty { get; set; }
+
+        public List<GraphProperty> properties;
+        public Dictionary<string, GraphProperty> overriddenProperties;
+
+        CSGTree tree;
+        List<Mesh> meshes;
+
+        void Start()
+        {
+            meshes = new List<Mesh>();
+            UpdateCSG();
+        }
+
+        void Update()
+        {
+            UpdateCSG();
```

Comment:
I assume UpdateCSG is here just for testing purposes? Calling Update methods doesn't scale, imagine 5000 graphs being run _every frame_. The overhead on calling "Update" will already be significant (unity calls "Update" through reflection = slow), whatever code is in UpdateCSG will just make this worse